File: 23eg112c21_py/Functions/palindrome.py
def pal(n):
    temp = n
    newno = 0
    while(n!=0):
        digit = n%10
        newno = newno*10+digit
        n = n//10
    if(temp == newno):
        return 1
    else:
        return 0
    

n = int(input("Enter a number: "))
if pal(n):
    print("Palindrome")
else:
    print("Not palindrome")

File: 23eg112c21_java/basic/palindrome.java
public class palindrome {
    public static void main(String [] args){
        int n = 125;
        int rev = 0;
        int tmp = n;
        while(tmp > 0){
            int digit = tmp % 10;
            rev = (rev * 10) + digit;
            tmp = tmp/10;
        }
        if(rev == n){
            System.out.println("Palindrome");
        }
        else{
            System.out.println("Not Palindrome");
        }
    }
}
